### PR TITLE
E2e tests for apps examples

### PR DESCRIPTION
**Pull request by Ukochka**
e2e tests for:
app-with-rbac
app-with-service-account
app-with-scenes

### CODE REVIEW

File: .github/workflows/integration-tests.yml [hunk 1]
```diff
@@ -46,7 +46,8 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        pluginDir: ${{ fromJson(needs.setup-matrix.outputs.pluginDirs) }}
+        #pluginDir: ${{ fromJson(needs.setup-matrix.outputs.pluginDirs) }}
+        pluginDir: ${{fromJson('["examples/app-with-service-account"]')}}
```
Comment (sunker) [inline .github/workflows/integration-tests.yml:50]: wasn't meant to be pushed?

File: .github/workflows/integration-tests.yml [hunk 2]
```diff
@@ -163,7 +167,7 @@ jobs:
       - name: Expected version - Start Grafana
         if: steps.has-integration-tests.outputs.DIR == 'true' && steps.should-run-expected-latest-tests.outcome == 'success'
         run: |
-          docker run -d -p 3000:3000 --add-host=host.docker.internal:host-gateway --name $PLUGIN_ID -v ${PWD}/dist:/var/lib/grafana/plugins/$PLUGIN_ID -v ${PWD}/provisioning:/etc/grafana/provisioning -e GF_DEFAULT_APP_MODE -e GF_INSTALL_PLUGINS -e GF_AUTH_ANONYMOUS_ORG_ROLE -e GF_AUTH_ANONYMOUS_ENABLED -e GF_AUTH_BASIC_ENABLED grafana/grafana:$EXPECTED_GRAFANA_VERSION
+          docker run -d -p 3000:3000 --add-host=host.docker.internal:host-gateway --name $PLUGIN_ID -v ${PWD}/dist:/var/lib/grafana/plugins/$PLUGIN_ID -v ${PWD}/provisioning:/etc/grafana/provisioning -e GF_DEFAULT_APP_MODE -e GF_INSTALL_PLUGINS -e GF_AUTH_ANONYMOUS_ORG_ROLE -e GF_AUTH_ANONYMOUS_ENABLED -e GF_AUTH_BASIC_ENABLED ${GITHUB_ENV:+-e GF_FEATURE_TOGGLES_ENABLE=${{ env.GF_FEATURE_TOGGLES_ENABLE }}} grafana/grafana:$EXPECTED_GRAFANA_VERSION
```
Comment (sunker) [inline .github/workflows/integration-tests.yml:170]: Think we can simplify this with docker compose. Try this: 
```
- name: Start Grafana
        run: |
          docker-compose pull
          docker-compose up -d
```

File: examples/app-with-rbac/.config/Dockerfile [hunk 3]
```diff
@@ -29,14 +29,14 @@ USER root
...
-      apt-get update && \
-      apt-get install -y supervisor inotify-tools git && \
-      rm -rf /var/lib/apt/lists/*; \
+    DEBIAN_FRONTEND=noninteractive && \
+    apt-get update && \
+    apt-get install -y supervisor inotify-tools git && \
+    rm -rf /var/lib/apt/lists/*; \
     else \
-      echo 'ERROR: Unsupported base image' && /bin/false; \
+    echo 'ERROR: Unsupported base image' && /bin/false; \
```
Comment (sunker) [inline examples/app-with-rbac/.config/Dockerfile:39]: What are these changes?

File: examples/app-with-rbac/package.json [hunk 4]
```diff
@@ -18,7 +19,10 @@
   "devDependencies": {
     "@babel/core": "^7.21.4",
     "@grafana/eslint-config": "^7.0.0",
+    "@grafana/plugin-e2e": "1.3.1",
```
Comment (sunker) [inline examples/app-with-rbac/package.json:22]: please use a range, and specify the latest version. should be ^1.5.0 as of now

File: examples/app-with-scenes/package.json [hunk 5]
```diff
@@ -19,6 +20,8 @@
     "@babel/core": "^7.21.4",
     "@grafana/eslint-config": "^7.0.0",
     "@grafana/tsconfig": "^1.2.0-rc1",
+    "@grafana/plugin-e2e": "1.3.1",
```
Comment (sunker) [inline examples/app-with-scenes/package.json:23]: same here

File: examples/app-with-service-account/package.json [hunk 6]
```diff
@@ -19,6 +20,8 @@
     "@babel/core": "^7.21.4",
     "@grafana/eslint-config": "^7.0.0",
     "@grafana/tsconfig": "1.3.0-rc1",
+    "@grafana/plugin-e2e": "1.3.1",
```
Comment (sunker) [inline examples/app-with-service-account/package.json:23]: and here

File: examples/app-with-service-account/tests/appRequests.spec.ts [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import { test, expect } from './fixtures';
+import { ROUTES } from '../src/constants';
+
+test.describe('app requests', () => {
```
Comment (sunker) [inline examples/app-with-service-account/tests/appRequests.spec.ts:4]: nit: not necessary to have the describe block when you only have one test

**Review by sunker — COMMENTED** [review]
Added some minor comments. Also please remove this file: `examples/app-with-service-account/tests/auth.setup.js`

For the scenes example app, I think we could do more specific tests. I have not yet explored how plugin-e2e can be tailored to better meet the need for scenes apps. If you want to, you can take a look at this. If now is not a good time for that, I'm fine just creating an issue for that so we can come back to this later on. Then we'll merge the scenes app tests as is.

**Comment by sunker** [comment]
Also remember not to merge this before [this](https://github.com/grafana/plugin-tools/pull/965) pr is merged and released.

**Comment by sunker** [comment]
@Ukochka when updating plugin-e2e, can you please bump all example plugins when you're at it? Also saw that a bunch of example plugin uses a pinned version of plugin-e2e. Can you change that to use the `^` range?

File: examples/app-with-rbac/package.json [hunk 8]
```diff
@@ -18,7 +19,10 @@
...
+    "@grafana/plugin-meta-extractor": "^0.0.2",
```
Comment (jackw) [inline examples/app-with-rbac/package.json:23]: How did this package end up in here? 🤔 Please remove it as it should not be used anywhere and will produce `generated` properties on plugin.json files which during a plugin submission will be flagged.

File: .github/workflows/integration-tests.yml (hunk 2 above)
Comment (Ukochka) [reply .github/workflows/integration-tests.yml]: I am not sure we can do it. we are running 3 different versions of grafana here: current, latest and canary. We can change it, but I prefer to make the separate task for it, if you do not mind.

**Comment by Ukochka** [comment]
@sunker I've fixed all the comment, please have a look.
About tests for scenes - I think we can improve tests in separate task, I can work on it in parallel with other tasks. But now I would like to finish this one and make automatic updates for this repo

File: .github/workflows/integration-tests.yml (hunk 2 above)
Comment (sunker) [reply .github/workflows/integration-tests.yml]: You'd just pass grafana version as an argument `GRAFANA_VERSION=$EXPECTED_GRAFANA_VERSION`. Then you wouldn't need that env file. 

this is not a blocker from my side. leaving it up to you

**Review by sunker — APPROVED** [review]
LGTM

**Pull request merged**